contracts/helpers/Harvester.sol

```diff
@@ -0,0 +1,185 @@
+            // Need to increase exposure to yield bearing asset
+            increase = 1;
+            amount = stablecoinsFromCollateral - targetExposureScaled / 1e9;
+            uint256 maxValueScaled = collatInfo.maxExposureYieldAsset * stablecoinsIssued;
+            // These checks assume that there are no transaction fees on the stablecoin->collateral conversion and so
+            // it's still possible that exposure goes above the max exposure in some rare cases
+            if (stablecoinsFromVault * 1e9 > maxValueScaled) amount = 0;
+            else if ((stablecoinsFromVault + amount) * 1e9 > maxValueScaled)
+                amount = maxValueScaled / 1e9 - stablecoinsFromVault;
+        } else {
+            // In this case, exposure after the operation might remain slightly below the targetExposure as less
+            // collateral may be obtained by burning stablecoins for the yield asset and unwrapping it
+            amount = targetExposureScaled / 1e9 - stablecoinsFromCollateral;
+            uint256 minValueScaled = collatInfo.minExposureYieldAsset * stablecoinsIssued;
+            if (stablecoinsFromVault * 1e9 < minValueScaled) amount = 0;
+            else if (stablecoinsFromVault * 1e9 < minValueScaled + amount * 1e9)
+                amount = stablecoinsFromVault - minValueScaled / 1e9;
+        }
+        if (amount > 0) {
+            try TRANSMUTER.updateOracle(collatInfo.vault) {} catch {}
+
+            rebalancer.adjustYieldExposure(
+                amount,
+                increase,
+                collateral,
+                collatInfo.vault,
+                (amount * (1e9 - maxSlippage)) / 1e9
+            );
+        }
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+                                                        SETTERS                                                     
+    //////////////////////////////////////////////////////////////////////////////////////////////////////////////////*/
+
+    function setRebalancer(address _newRebalancer) external onlyGuardian {
+        if (_newRebalancer == address(0)) revert ZeroAddress();
+        rebalancer = RebalancerFlashloan(_newRebalancer);
+    }
+
+    function setCollateralData(
+        address vault,
+        uint64 targetExposure,
+        uint64 minExposureYieldAsset,
+        uint64 maxExposureYieldAsset,
+        uint64 overrideExposures
+    ) external onlyGuardian {
+        _setCollateralData(vault, targetExposure, minExposureYieldAsset, maxExposureYieldAsset, overrideExposures);
+    }
+
+    function setMaxSlippage(uint96 _maxSlippage) external onlyGuardian {
+        _setMaxSlippage(_maxSlippage);
+    }
+
+    function updateLimitExposuresYieldAsset(address collateral) external {
+        CollatParams storage collatInfo = collateralData[collateral];
+        if (collatInfo.overrideExposures == 0) _updateLimitExposuresYieldAsset(collatInfo);
```

Comment:
This means you can setup collateral Data for unintialised collateral.
Let's have another number for default